client/my-sites/plans/main.jsx

```diff
@@ -111,14 +113,24 @@ class Plans extends React.Component {
 	}
 }
 
-export default connect( state => {
+export const mapStateToProps = ( state, ownProps ) => {
 	const selectedSiteId = getSelectedSiteId( state );
 
 	const jetpackSite = isJetpackSite( state, selectedSiteId );
 	const isSiteAutomatedTransfer = isSiteAutomatedTransferSelector( state, selectedSiteId );
+	let intervalType = ownProps.intervalType;
+	if ( ! intervalType && config.isEnabled( 'upgrades/2-year-plans' ) ) {
```

Comment:
This change makes `intervalType` a passed-in prop _and_ a connected prop? That seems like a confusing pattern. Is it possible to make it either one or the other?

Or, to rephrase my question, if we can use the selector here to get the interval type, why do we still need to pass in the prop from above?

Reply:
> This prop comes form the URL. The URL pattern for plans controller is /plans/:intervalType?/:site.

Ah, thanks!

> Can you please elaborate on why is that confusing?

Usually I would expect there to be just one source for a prop, and this override adds some complexity to what is already a complex component. I understand the use case better now, though.

Exploring this further, when using this form currently:
``` 
/plans/:intervalType?/:site
```
with no `intervalType` specified,  is the display expected to default to one particular interval, and if so, which? And, will this PR change that behaviour?